glean-core/ios/Glean/Net/HttpPingUploader.swift

```diff
@@ -4,6 +4,27 @@
 
 import Foundation
 
+/// `true` to allow the uploader to process pings.
+/// Note that this does not mean that an uploader is actually running.
+/// It will be invoked when a ping is submitted.
+///
+/// `false` to stop the uploader.
+var stateRunning: AtomicBoolean = AtomicBoolean(false)
+
+/// The uploader will enter and leave this group as it starts and finishes work.
+let uploadGroup = DispatchGroup()
+
+func shutdownUploader() {
+    stateRunning.value = false
+    // Wait for an uploader to finish it's current upload.
+    // If no uploader is running, this returns immediately.
+    uploadGroup.wait()
+}
+
+func startUploader() {
+    stateRunning.value = true
+}
```

Comment:
oh this whole thing? more mutable global state!
I'm considering moving this into `HttpPingUploader` below and have it be a singleton like `Glean`